### PR TITLE
Remove i18n-inflector-rails dependency

### PR DESCRIPTION
**Pull request by SuperTux88**
There are no new releases anymore and the current version isn't compatible with ruby 3.x, see: https://github.com/siefca/i18n-inflector-rails/pull/10 and https://github.com/siefca/i18n-inflector-rails/pull/11

As this feature wasn't really used a lot (Icelandic didn't even setup inflections properly), it's probably not worth fighting for it, so lets just drop it.

Related to #8369

### CODE REVIEW

**Comment by denschub** [comment]
I feel like leaving a comment here just for the after-world if this issue resolves itself. This is too funny to let go.

The tests for this PR are currently red:

Screenshot 2023-06-09 at 03 44 47

Why, you ask? Well:

Screenshot 2023-06-09 at 03 45 30

Because our test suite somehow decided that it now wants to run in French instead of English. Why does it do that? We have absolutely no idea.

**Comment by Flaburgan** [comment]
> Because our test suite somehow decided that it now wants to run in French instead of English. Why does it do that? We have absolutely no idea.

We run the world :joy:

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -91,9 +88,11 @@ def set_diaspora_header
   def set_locale
     if user_signed_in?
       I18n.locale = current_user.language
+      puts "logged in locale: #{locale}"
```
Comment (diaspora-code-review) [inline app/controllers/application_controller.rb:91]: Rails/Output: Do not write to stdout. Use Rails's logger if you want to log.

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -91,9 +88,11 @@ def set_diaspora_header
...
     else
       locale = http_accept_language.language_region_compatible_from AVAILABLE_LANGUAGE_CODES
       locale ||= DEFAULT_LANGUAGE
+      puts "locale: #{locale}"
```
Comment (diaspora-code-review) [inline app/controllers/application_controller.rb:95]: Rails/Output: Do not write to stdout. Use Rails's logger if you want to log.

**Comment by SuperTux88** [comment]
> Because our test suite somehow decided that it now wants to run in French instead of English. Why does it do that? We have absolutely no idea.

OK, as fun as this sounds, it was actually very hard to debug and find out what's going on (especially because whatever I tried, I wasn't able to reproduce it locally). But I finally found the reason ... it turns out that after the locale was set to French once, it modified the English default translations, and after that point English was translated to French in-memory ... so lets just not do that anymore ...

I have no idea how exactly that worked before, maybe something with the inflector prevented that from happening? But I already debugged enough today, so I'm just happy it works now.

**Review by jhass — APPROVED** [review]

**Comment by denschub** [comment]
Amazing investigative work, @SuperTux88. 🙌

**Pull request merged**